opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -122,7 +201,47 @@ public function index() {
         );
         $this->response->setOutput($this->render());
     }
-
+    protected function handleCustomerSync() {
+        $isRequestMethodPost = $this->request->server['REQUEST_METHOD'] === 'POST';
+        $hasPermissionsToModify = $this->user->hasPermission('modify', 'module/smaily_for_opencart');
+        if (!$isRequestMethodPost || !$hasPermissionsToModify) {
+            return;
+        }
+        $response = [];
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Declare customer sync objects.
+        $subscribe_status = $this->request->post['smaily_for_opencart_enable_subscribe'];
+        // For if no customer sync fields are selected.
+        if (!empty($this->request->post['smaily_for_opencart_syncronize_additional'])) {
+            $subscribe_additional = $this->request->post['smaily_for_opencart_syncronize_additional'];
+        } else {
+            $subscribe_additional = array();
+        }
+        $subscribe_sync_token = $this->request->post['smaily_for_opencart_sync_token'];
+        
+        // Data validation.
+        $subscribe_status = ($subscribe_status == 0 || $subscribe_status == 1) ? $subscribe_status : 0;
+        // Flipping array allows to use isset for validation.
+        $subscribe_additional = (
```

Comment:
This sanitization would still allow someone to post fields `['firstname', 'some_other_field']`. A better way would be to just ignore fields that do not exist:
```php
$subscribe_additional = array_intersect(['firstname', 'lastname', 'telephone', 'date_added'], $subscribe_additional);
```